src/core/xds/grpc/xds_cluster.h

```diff
@@ -81,6 +81,19 @@ struct XdsClusterResource : public XdsResourceType::ResourceData {
     }
   };
 
+  struct UpstreamTlsContext {
+    CommonTlsContext common_tls_context;
+    std::string sni;
+    bool auto_host_sni;
```

Comment:
These two fields should default to false, so that if the env var is not enabled, they get the right values.

Reply:
Should `sni` also have a default value?

Reply:
Explicit initialization of fields is generally needed only for POD types, since C++ doesn't do that automatically, which can lead to use-of-uninitialized-variable bugs.

In contrast, class types generally initialize themselves deterministically at construction.  For `std::string`, the default value is an empty string; for `std::optional<>`, the default value is nullopt; for `std::unique_ptr<>`, the default value is nullptr; etc.